### PR TITLE
Convert build pipeline to 1ES

### LINKED ISSUE #1062: Convert pipelines to 1ES
Build pipeline for this repo must be converted to 1ES template for security purposes

### PR DESCRIPTION
**Pull request by andystaples**
### Issue describing the changes in this PR

resolves #1062
Creates #1065 

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [x] Otherwise: Backport tracked by issue/PR #1064 
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: eng/ci/official.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+resources:
+  repositories:
+    - repository: 1es
+      type: git
+      name: 1ESPipelineTemplates/1ESPipelineTemplates
+      ref: refs/tags/release
+
+variables:
+  Configuration: Release
+  buildNumber: $[ counter('build', 4000) ] # Start higher than our AppVeyor versions. Every build (pr or branch) will increment.
```
Comment (Francisco-Gamino) [inline eng/ci/official.yml]: Please update this comment.

File: eng/ci/templates/build.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+jobs:
+  - job:
+    templateContext:
+      outputs:
+        - output: nuget
+          packagesToPush: "$(Build.ArtifactStagingDirectory)/*.nupkg"
+          packageParentPath: "$(Build.ArtifactStagingDirectory)"
+          nuGetFeedType: internal
+          publishVstsFeed: "e6a70c92-4128-439f-8012-382fe78d6396/c0493cce-bc63-4e11-9fc9-e7c45291f151"
+          # sbomBuildDropPath: '$(System.DefaultWorkingDirectory)/package'
```
Comment (Francisco-Gamino) [inline eng/ci/templates/build.yml]: If we do not need this, please remove them.

File: eng/ci/templates/build.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          sbomPackageName: "Azure Functions PowerShell Worker"
+          sbomBuildComponentPath: "$(Build.SourcesDirectory)"
+          allowPackageConflicts: true
+        # - output: nuget
```
Comment (Francisco-Gamino) [inline eng/ci/templates/build.yml:14]: The same for these.

**Review by Francisco-Gamino — APPROVED** [review]
I left two minor comments. Otherwise, LGTM.

File: eng/ci/official.yml (hunk 1 above)
Comment (andystaples) [reply eng/ci/official.yml]: Done

File: eng/ci/templates/build.yml (hunk 2 above)
Comment (andystaples) [reply eng/ci/templates/build.yml]: Done

**Pull request merged**